### PR TITLE
Update security docs for seccomp/apparmor

### PR DESCRIPTION
**Pull request by moxiegirl**
Carry of docs for: #17989 

Signed-off-by: Mary Anthony mary@docker.com

### CODE REVIEW

**Comment by moxiegirl** [comment]
Not ready for review. I'll switch the label when it is.

**Comment by moxiegirl** [comment]
ping @jfrazelle @thaJeztah

File: docs/security/seccomp.md [hunk 1]
```diff
@@ -3,27 +3,25 @@
...
-Passing a profile for a container
----------------------------------
-
-Users may pass a seccomp profile using the `security-opt` option
-(per-container).
-
-The profile has layout in the following form:
+The default seccomp profile provides a sane default for running containers with
+seccomp. It is moderately protective while providing wide application
+compatibility. The default Docker profile has layout in the following form:
```
Comment (thaJeztah) [inline docs/security/seccomp.md]: @jfrazelle does the example format below need to be updated if we switch to a whitelist? (https://github.com/docker/docker/pull/18979), I.e. will `"defaultAction": "SCMP_ACT_ALLOW"` be a "deny"?

Comment (jfrazelle) [inline docs/security/seccomp.md]: The user can decide either way, if a custom profile is run we don't use the default profile

Comment (jfrazelle) [inline docs/security/seccomp.md]: So it's unrelated :)

File: docs/security/apparmor.md [hunk 2]
```diff
@@ -1,47 +1,71 @@
...
 +++
 <![end-metadata]-->
 
-AppArmor security profiles for Docker
---------------------------------------
+# AppArmor security profiles for Docker
 
-AppArmor (Application Armor) is a security module that allows a system
-administrator to associate a security profile with each program. Docker
+AppArmor (Application Armor) is a Linux security module that protects an operating system and its applications from security threats. To use it, a system administrator associates an AppArmor security profile with each program. Docker
```
Comment (thaJeztah) [inline docs/security/apparmor.md]: nit: can you wrap this?

File: docs/security/seccomp.md (hunk 1 above)
Comment (thaJeztah) [inline docs/security/seccomp.md]: Well, I thought because it says "the default profile uses this format"..

**Comment by thaJeztah** [comment]
Should we write up how people can write their own profile for a container? I.e., how to find what to block/allow for a container to run? Doesn't have to be in this PR, just thinking that that would be something I would be looking for as a user.

Looks like the links checker found broken links (not at my computer now, so haven't tested them);
image

**Comment by jfrazelle** [comment]
Oh sorry am looking from email.. Will take a better look tomorrow

File: docs/security/seccomp.md (hunk 1 above)
Comment (jfrazelle) [inline docs/security/seccomp.md]: oh yes this would need to change but for now... it is correct so if we get this in i can update it in that PR :)

Comment (thaJeztah) [inline docs/security/seccomp.md]: sgtm

**Comment by jfrazelle** [comment]
Ya i am working on some stuff with regard to how to debug and writing your own, but this LGTM

File: docs/articles/security.md [hunk 3]
```diff
@@ -277,8 +277,9 @@ containerization systems, these are simply kernels features that may
 be implemented in Docker as well. We welcome users to submit issues,
 pull requests, and communicate via the mailing list.
 
-References:
+## Related Information
 
-* [Docker Containers: How Secure Are They? (2013)](
-http://blog.docker.com/2013/08/containers-docker-how-secure-are-they/).
-* [On the Security of Containers (2014)](https://medium.com/@ewindisch/on-the-security-of-containers-2c60ffe25a9e).
+* [Use trusted images](trust/index.md)
```
Comment (thaJeztah) [inline docs/articles/security.md]: These should be the correct links:

```Markdown
* [Use trusted images](../security/trust/index.md)
* [Seccomp security profiles for Docker](../security/seccomp.md)
* [AppArmor security profiles for Docker](../security/apparmor.md).
* [On the Security of Containers (2014)](https://medium.com/@ewindisch/on-the-security-of-containers-2c60ffe25a9e)
```

**Comment by thaJeztah** [comment]
@moxiegirl can you fix the links, then LGTM

File: docs/security/apparmor.md [hunk 4]
```diff
@@ -1,47 +1,71 @@
 <!-- [metadata]>
 +++
-draft = true
+title = "AppArmor security profiles for Docker"
+description = "Enabling seccomp in Docker"
```
Comment (justincormack) [inline docs/security/apparmor.md]: description is wrong here

Comment (thaJeztah) [inline docs/security/apparmor.md]: Good catch! I should pay more attention to the meta-data :)

**Comment by moxiegirl** [comment]
Updated, thanks for all the helpful comments!

**Comment by icecrime** [comment]
@moxiegirl Sorry, needs a rebase :panda_face:

**Comment by moxiegirl** [comment]
@thaJeztah wanna take a last minute peek at this after make sure it looks ok to you after the changes?

**Comment by justincormack** [comment]
@moxiegirl the diffs are not very easy to read but happy to do an extra review after merge if you want an extra pair of eyes.

File: docs/security/seccomp.md [hunk 5]
```diff
@@ -3,27 +3,25 @@
...
 
-Seccomp security profiles for Docker
-------------------------------------
+# Seccomp security profiles for Docker
 
-The seccomp() system call operates on the Secure Computing (seccomp)
-state of the calling process.
+Secure computing mode (Seccomp) is a Linux kernel feature. You can use it to
+restrict the actions available within the container.  The `seccomp()`
+system call operates on the seccomp state of the calling process. You can use this feature to restrict the access of your application.
```
Comment (jamtur01) [inline docs/security/seccomp.md]: Wrap line?

File: docs/security/seccomp.md [hunk 6]
```diff
@@ -3,27 +3,25 @@
...
+## Passing a profile for a container
 
-Passing a profile for a container
----------------------------------
-
-Users may pass a seccomp profile using the `security-opt` option
-(per-container).
-
-The profile has layout in the following form:
+The default seccomp profile provides a sane default for running containers with
```
Comment (jamtur01) [inline docs/security/seccomp.md]: Seccomp

File: docs/security/seccomp.md [hunk 7]
```diff
@@ -3,27 +3,25 @@
...
+seccomp. It is moderately protective while providing wide application
```
Comment (jamtur01) [inline docs/security/seccomp.md]: Seccomp

File: docs/security/seccomp.md (hunk 1 above)
Comment (jamtur01) [inline docs/security/seccomp.md]: Perhaps?

    The default Docker profile is laid out in the following form:

File: docs/security/seccomp.md [hunk 8]
```diff
@@ -143,3 +125,13 @@ the reason each syscall is blocked rather than white-listed.
 | `ustat`             | Obsolete syscall.                                                                                                                     |
 | `vm86`              | In kernel x86 real mode virtual machine. Also gated by `CAP_SYS_ADMIN`.                                                               |
 | `vm86old`           | In kernel x86 real mode virtual machine. Also gated by `CAP_SYS_ADMIN`.                                                               |
+
+## Run without the default seccomp profile
+
+You can pass `unconfined` to run a container without the default seccomp
```
Comment (jamtur01) [inline docs/security/seccomp.md]: Seccomp

File: docs/security/security.md [hunk 9]
```diff
@@ -277,8 +278,9 @@ containerization systems, these are simply kernels features that may
...
 
-References:
+## Related Information
 
-* [Docker Containers: How Secure Are They? (2013)](
-http://blog.docker.com/2013/08/containers-docker-how-secure-are-they/).
-* [On the Security of Containers (2014)](https://medium.com/@ewindisch/on-the-security-of-containers-2c60ffe25a9e).
+* [Use trusted images](../security/trust/index.md)
+* [Seccomp security profiles for Docker](../security/seccomp.md)
+* [AppArmor security profiles for Docker](../security/apparmor.md)
```
Comment (thaJeztah) [inline docs/security/security.md]: any reason for adding `../security/` to each of these?

File: docs/security/seccomp.md (hunk 6 above)
Comment (moxiegirl) [inline docs/security/seccomp.md]: It is seccomp unless at the start of a sentence or in a heading.

File: docs/security/seccomp.md (hunk 7 above)
Comment (moxiegirl) [inline docs/security/seccomp.md]: see previous note.

File: docs/security/seccomp.md (hunk 8 above)
Comment (moxiegirl) [inline docs/security/seccomp.md]: see previous note.

File: docs/security/security.md (hunk 9 above)
Comment (moxiegirl) [inline docs/security/security.md]: Because they are required for the links to work. `/security/seccomp.md` and `security/seccomp.md` fail.

**Comment by moxiegirl** [comment]
@thaJeztah Go! Seb Go!

**Comment by thaJeztah** [comment]
LGTM!! \o/

File: docs/security/security.md (hunk 9 above)
Comment (thaJeztah) [inline docs/security/security.md]: `seccomp.md` worked though :)
anyway, just a nit, so no worries, haha

**Pull request merged**